### PR TITLE
Issue #5 - Automating Password Entry

### PR DESCRIPTION
**Pull request by name-schema**
In commands/copy.go
  Created new function getPasswordFromEnv to work as a fallback
   password if the -p flag is not set. Important to note, the -p
   flag takes priority over the env var, as it should.
  Updated logic to check password length before using it to encrypt

In commands/env.go
  Added new env const:
   const pbgopyPasswordFileEnv = "PBGOPY_PASSWORD_FILE"

In commands/paste.go
  Updated logic to check password length before using it to decrypt

Fixes: https://github.com/nakabonne/pbgopy/issues/5

### CODE REVIEW

**Comment by nakabonne** [comment]
Thanks for your contribution! I'm in the middle of something, I'll take a look later!

**Comment by jcwillox** [comment]
Just an idea, why not store the password in an environment variable like `PBGOPY_TOKEN` or `PBGOPY_PASS`.

File: commands/copy.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ func NewCopyCommand(stdout, stderr io.Writer) *cobra.Command {
 }
 
 func (r *copyRunner) run(_ *cobra.Command, _ []string) error {
+
```
Comment (nakabonne) [inline commands/copy.go:44]: Please remove the unneeded line.

File: commands/copy.go [hunk 2]
```diff
@@ -53,12 +54,24 @@ func (r *copyRunner) run(_ *cobra.Command, _ []string) error {
...
 	if r.password != "" {
+		password = r.password
+	} else if os.Getenv(pbgopyPasswordFileEnv) != "" {
+		password, err = getPasswordFromEnv(os.Getenv(pbgopyPasswordFileEnv))
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(password) != 0 {
```
Comment (nakabonne) [inline commands/copy.go:69]: Why not just compare with "".
```suggestion
	if password != "" {
```

File: commands/copy.go [hunk 3]
```diff
@@ -74,6 +87,27 @@ func (r *copyRunner) run(_ *cobra.Command, _ []string) error {
 	return nil
 }
 
+func getPasswordFromEnv(fileLoc string) (string, error) {
```
Comment (nakabonne) [inline commands/copy.go:90]: nit
```suggestion
func getPasswordFromEnv(filepath string) (string, error) {
```

File: commands/paste.go [hunk 4]
```diff
@@ -57,12 +57,24 @@ func (r *pasteRunner) run(_ *cobra.Command, _ []string) error {
...
 	if r.password != "" {
+		password = r.password
+	} else if os.Getenv(pbgopyPasswordFileEnv) != "" {
+		password, err = getPasswordFromEnv(os.Getenv(pbgopyPasswordFileEnv))
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(password) != 0 {
```
Comment (nakabonne) [inline commands/paste.go:72]: Here as well:
```suggestion
	if password != "" {
```

**Review by nakabonne — CHANGES_REQUESTED** [review]

**Comment by nakabonne** [comment]
@name-schema Thank you! Left some comments.

**Comment by nakabonne** [comment]
@jcwillox Nice idea. Can you open an issue for that?

**Comment by nakabonne** [comment]
And I'm sorry but could you resolve the conflicts?

**Comment by name-schema** [comment]
Working on that today, was out of pocket.

**Comment by nakabonne** [comment]
Thank you! I'm in the middle of something, so will take a look this weekend!

**Review by nakabonne — APPROVED** [review]
Thank you!

**Pull request merged**